docs/source/en/kernel_doc/overview.md

```diff
@@ -1,3 +1,171 @@
-# Overview
+# Hugging Face Transformers Kernels
 
-Kernels in transformers are used to optimize the performance of models with custom layers from the hub and very low effort.
+The Transformers Kernels integration provides high-performance, optimized kernel implementations for common transformer operations. By leveraging specialized CUDA, Triton, ROCm, Metal and XPU kernels from the Hugging Face Hub, you can significantly accelerate model inference and training with minimal code changes.
+
+## Key Benefits
+
+- **Drop-in Performance Boost**: Enable optimized kernels with a single `use_kernels=True` flag
+- **Hub-Based Distribution**: Access community-maintained kernels directly from Hugging Face Hub
+- **Multi-Backend Support**: CUDA, ROCm, Metal, and XPU backends for different hardware platforms
+- **Mode-Aware Optimization**: Automatically switches between training and inference optimizations
+- **Zero Code Changes**: Existing models automatically benefit from kernel acceleration
+- **Customizable**: Override default kernels with your own implementations via `KernelConfig`
+
+## Supported Operations
+
+The `kernels` library provides optimized implementations for:
+
+### Normalization Layers
+- **RMSNorm**: Root Mean Square Layer Normalization
+
+### Activation Functions
+- **FastGELU, NewGELU, QuickGELU**: GELU variants
+- **SiLU**: Sigmoid Linear Unit
+- **GeluTanh**: GELU with Tanh approximation
+
+### MLP and MoE Layers
+- **MLP**: Standard Multi-Layer Perceptron layers
+- **MegaBlocksMoeMLP**: Optimized Mixture-of-Experts implementations
+- **Llama4TextMoe**: Llama 4 MoE layers
+
+### Attention Mechanisms
+- **Flash Attention**: Fast attention implementations
+- **MultiScaleDeformableAttention**: For vision transformers
+- **Custom Attention**: Load community kernels for specialized attention patterns
+
+### Specialized Operations
+- **Mamba Selective Scan**: Built-in optimized kernels for Mamba SSM models
+- **Causal Convolution 1D**: Efficient causal convolutions
+
+## Quick Start
+
+### Basic Usage
+
+Enable kernels when loading any model:
```

Comment:
Clarify that this means the optimized kernels are automatically pulled in and used